pkgs/development/compilers/dtc/default.nix

```diff
@@ -110,11 +110,23 @@ stdenv.mkDerivation (finalAttrs: {
     # https://github.com/NixOS/nixpkgs/pull/118700#issuecomment-885892436
     !stdenv.isDarwin &&
 
+    # tests attempt to link .so files
+    # https://github.com/dgibson/dtc/commit/dcef5f834ea34bcb9f8d0e86db1268fde52ead77
+    !stdenv.hostPlatform.isStatic &&
+
     # we must explicitly disable this here so that mesonFlags receives
     # `-Dtests=disabled`; without it meson will attempt to run
     # hostPlatform binaries during the configurePhase.
     (with stdenv; buildPlatform.canExecute hostPlatform);
 
+  strictDeps = true;
+
+  nativeCheckInputs = [ pkg-config ];
+  preCheck = ''
+    # remove on next release: https://github.com/dgibson/dtc/commit/dcef5f834ea34bcb9f8d0e86db1268fde52ead77
```

Comment:
Can we backport the patch instead?

Reply:
patch does not apply as is. Do you prefer an in-tree, rebased copy of the patch?

Reply:
In that case this is fine by mee.